src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -1077,6 +1079,13 @@ bool GCToEEInterface::EagerFinalized(Object* obj)
         FinalizeWeakReference(obj);
         return true;
     }
+#ifdef FEATURE_OBJCBRIDGE
+    else if (pMT->IsTrackedReferenceWithFinalizer())
+    {
+        ObjCBridgeNative::OnEnteredFinalizerQueue((OBJECTREF)obj);
+        return false;
```

Comment:
returning false will allow finalizer to run. That can resurrect the object and/or re-arm it for finalization. Are we ok with that?

Reply:
Another weird thing that could happen is that finalization could be suppressed for an object. That is often done when resources could be freed in some other way. 
`EagerFinalized` does not respect the suppression though. So you may get in a situation where this code will run, but finalizer will not. Not sure if that is a concern or actually a good thing, just something to think about.

Reply:
Yes, that is intentional. We are using this as the hook into adding to the finalizer queue.